lib/types.bzl

```diff
@@ -138,6 +138,20 @@ def _is_set(v):
     """
     return type(v) == _a_struct_type and hasattr(v, "_values") and _is_dict(v._values)
 
+def _is_partial(v):
+    """Returns True if v is a partial function object created by partial.make().
+
+    Args:
+      v: The value whose type should be checked.
+
+    Returns:
+      True if v was created by partial.make(), False otherwise.
+    """
+    return type(v) == _a_struct_type \
```

Comment:
This is only a heuristic. I wonder if making the partial struct an instance of a distinct provider might allow a more robust check.

FWIW, partial is hack to work around the fact that nested def statements don't currently work, but I would like to fix that, at which point we can get start to rid of partial.

Reply:
I was following the precedent and model of `_is_set()` just above, since it, too, pertains to a skylib-specific type.
Note that `_is_partial()` is a _macro_ and gets used by other macros in skylib, in particular, in `_suite()` in `lib/unittest.bzl`.  My understanding is that providers are available only within _rules_, so they would not be available for use by macros.

Is there an example that you could point me to that would illustrate the alternative implementation you're suggesting?

Reply:
Will do. Would you like the name to remain `is_partial`?  That way the usage sites would look like `partial.is_partial()`, which strikes me as a bit redundant.  Another possibility that occurs to me is to call it `made`, since it checks whether something came from `make`. That way the usage sites would look like `partial.made()`, to check if something came from `partial.make()`.

Reply:
`partial.is_instance(x)`?  `made` seems a little too cryptic.

Reply:
Sounds good, I'll use `is_instance()`.

Now, is it okay to make `partial.is_instance()` use the predicates defined in `types.bzl`? Or should I code it so that `partial.bzl` does not do any additional `load()`s?

Reply:
I will err on the side of being self-contained. If you'd prefer to keep things DRY at the cost of making `partial.bzl` depend on `types.bzl`, please just let me know.

Reply:
Okay, I believe I've addressed all your comments so far.